### PR TITLE
[AOT] fix type in debug.ts

### PR DESCRIPTION
**Pull request by PPpro**
Re: https://github.com/cocos/3d-tasks/issues/16413

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/platform/debug.ts [hunk 1]
```diff
@@ -49,80 +50,68 @@ let ccAssert = (condition: any, message?: any, ...optionalParams: any[]): void =
...
- * @zh 输出一条消息到 Cocos Creator 编辑器的 Console 或运行时 Web 端的 Console 中。这为你提供了对输出格式的额外控制。
- * @param message @zh 包含零个或多个需要替换的JavaScript字符串。@en JavaScript objects to replace substitution strings in msg.
- * @param optionalParams  @zh 用来替换在message中需要替换的JavaScript对象。@en JavaScript objects with which to replace substitution strings within msg.
+ * @en Outputs a log message to the console. The message may be a single string (with optional substitution values), or it may be any one or more JavaScript objects.
+ * @zh 向控制台输出一条日志信息。这条信息可能是单个字符串（包括可选的替代字符串），也可能是一个或多个对象。
  */
-export function log (message?: any, ...optionalParams: any[]): void {
-    return ccLog(message, ...optionalParams);
+export function log (...data: unknown[]): void {
+    return ccLog(...data);
```
Comment (PPpro) [inline cocos/core/platform/debug.ts:64]: Our implementation is aligned to web standards, so the `log` method should be defined as `function log (...data: unknown[]): void`, the same goes for other methods like `warn error debug`

File: cocos/core/utils/js-typed.ts [hunk 2]
```diff
@@ -348,7 +348,21 @@ const REGEXP_STR = /%s/;
...
+ * ```
+ * import { js } from 'cc';
+ * js.formatStr({}, null, undefined);  // [object Object] null undefined
+ * ```
+ */
+export function formatStr (...data: unknown[]): string;
+export function formatStr (msg: unknown, ...subst: unknown[]): string {
     if (arguments.length === 0) {
         return '';
     }
```
Comment (PPpro) [inline cocos/core/utils/js-typed.ts:368]: we should support another formatStr signature, see this use case in unit test:
https://github.com/cocos/cocos-engine/blob/2a5b5711c1692c235bf05275b73cabacc9b15503/tests/core/js.test.ts#L41

File: cocos/2d/components/ui-mesh-renderer.ts [hunk 3]
```diff
@@ -268,7 +268,7 @@ export class UIMeshRenderer extends Component {
      */
     get renderEntity (): RenderEntity {
         if (DEBUG) {
-            assert(this._renderEntity, 'this._renderEntity should not be invalid');
+            assert(!!this._renderEntity, 'this._renderEntity should not be invalid');
```
Comment (minggo) [inline cocos/2d/components/ui-mesh-renderer.ts]: Why use `!!` as it is not clear to be understood.

Comment (PPpro) [reply cocos/2d/components/ui-mesh-renderer.ts]: it means we convert this object to a boolean value

Comment (minggo) [reply cocos/2d/components/ui-mesh-renderer.ts]: It is better to make it more clear, for example: `this._renderEntity != null`

File: cocos/core/utils/js-typed.ts (hunk 2 above)
Comment (minggo) [reply cocos/core/utils/js-typed.ts:368]: I don't quite understand this change. What problem does it fix?

File: cocos/core/platform/debug.ts (hunk 1 above)
Comment (minggo) [reply cocos/core/platform/debug.ts:64]: Will it break compatibility?

File: cocos/core/utils/js-typed.ts (hunk 2 above)
Comment (PPpro) [reply cocos/core/utils/js-typed.ts:368]: Before, we only supported `formatStr (msg: string, ...subst: StringSubstitution[]): string;`, in reality, the `log` method can pass in any js object, not just a string, such as the following :

```ts
log(null)
log(undefined)
log({}, 1, 2, 3)
```

In this situation, `function formatStr (msg: string, ...subst: StringSubstitution[]): string` can no longer meet the demand, we need a `function formatStr (...data: unknown[]): string` type declaration to support constructing any js object sequence into a string, in fact, the implementation of `formatStr` itself can support this situation, it just lacks a type declaration

Comment (PPpro) [reply cocos/core/utils/js-typed.ts:368]: In addition, the definition `formatStr (msg: string, ...subst: StringSubstitution[]): string` does not satisfy the usage of this test case:
https://github.com/cocos/cocos-engine/blob/2a5b5711c1692c235bf05275b73cabacc9b15503/tests/core/js.test.ts#L41

File: cocos/2d/components/ui-mesh-renderer.ts (hunk 3 above)
Comment (PPpro) [reply cocos/2d/components/ui-mesh-renderer.ts]: We may also need to judge the situation of `undefined`, which will be more verbose.
I'll consider to use `Boolean(this._renderEntity)`

File: cocos/core/utils/js-typed.ts (hunk 2 above)
Comment (PPpro) [reply cocos/core/utils/js-typed.ts:368]: With the declaration before we can only support the following use cases:
```ts
log('hello world')
// or
log('hello %s', 'world')
```

File: cocos/core/platform/debug.ts (hunk 1 above)
Comment (PPpro) [reply cocos/core/platform/debug.ts:64]: no, it won't, any variable can be assigned to a variable of type `unknown`

and `unknown` is safer than `any`

File: cocos/2d/components/ui-mesh-renderer.ts (hunk 3 above)
Comment (PPpro) [reply cocos/2d/components/ui-mesh-renderer.ts]: updated https://github.com/cocos/cocos-engine/pull/15426/commits/52ab100e94922a98c6d05dab24257f1faedea506

**Review by minggo — APPROVED** [review]

**Pull request merged**